shell/shared/grain.js

```diff
@@ -273,32 +280,70 @@ Meteor.methods({
+          try {
+            SandstormEmail.send({
+              to: emailAddress,
+              from: "Sandstorm server <no-reply@" + HOSTNAME + ">",
+              subject: sharerDisplayName + " has invited you to join a grain: " + title,
+              text: message.text + "\n\nFollow this link to open the shared grain:\n\n" + url +
+                "\n\nNote: If you forward this email to other people, they will be able to " +
+                "access the share as well. To prevent this, remove the link before forwarding.",
+              html: html,
+            });
+          } catch (e) {
+            outerResult.failures.push({contact: contact, error: e.toString()});
+          }
+        } else {
+          SandstormPermissions.createNewApiToken(
+            globalDb, {identityId: identityId, accountId: accountId}, grainId,
+            "direct invitation to " + contact.profile.intrinsicName,
+            roleAssignment, {user: {identityId: contact._id, title: title}});
+            try {
+              var identity = Meteor.users.findOne({_id: contact._id});
+              var emailAddress = SandstormDb.getVerifiedEmails(identity)[0];
+              var url = origin + "/grain/" + grainId;
+              var html = message.html + "<br><br>" +
+                  "<a href='" + url + "' style='display:inline-block;text-decoration:none;" +
+                  "font-family:sans-serif;width:200px;min-height:30px;line-height:30px;" +
+                  "border-radius:4px;text-align:center;background:#428bca;color:white'>" +
+                  "Open Shared Grain</a><div style='font-size:8pt;font-style:italic;color:gray'>" +
+                  "Note: This was shared directly with you. No one else can open the above " +
+                  "link, and you will need to be logged in with your "+  contact.profile.service +
+                  " account with username " + contact.profile.intrinsicName;
+              if (emailAddress) {
+                SandstormEmail.send({
+                  to: emailAddress,
+                  from: "Sandstorm server <no-reply@" + HOSTNAME + ">",
+                  subject: sharerDisplayName + " has invited you to join a grain: " + title,
+                  text: message.text + "\n\nFollow this link to open the shared grain:\n\n" + url +
+                    "\n\nNote: This was shared directly with you. No one else can open the above " +
+                    "link, and you will need to be logged in with your " + contact.profile.service +
+                    " account with username " + contact.profile.intrinsicName,
+                  html: html,
+                });
+              } else {
+                outerResult.failures.push({contact: contact, warning: "User does not have a " +
+                  "verified email, so notification of this share was not sent to them. Please " +
+                  "manually share " + url + " with them."});
+              }
+            } catch (e) {
```

Comment:
If something goes wrong with sending the email (e.g. the SMTP URL is not properly configured), it looks like this will display the error about the email and possibly leave the impression that the share failed, whereas it was actually just the notification that failed.